src/plugins/ssh/index.ts

```diff
@@ -92,60 +88,6 @@ const UNPROVISIONED_ACCESS_MESSAGES = [
   { pattern: DESTINATION_READ_ERROR },
 ];
 
-/** Checks if access has propagated through AWS to the SSM agent
- *
- * AWS takes about 8 minutes, GCP takes under 1 minute
- * to fully resolve access after it is granted.
- * During this time, calls to `aws ssm start-session` / `gcloud compute start-iap-tunnel`
- * will fail randomly with an various error messages.
- *
- * This function checks the subprocess output to see if any of the error messages
- * are printed to the error output within the first 5 seconds of startup.
- * If they are, the returned `isAccessPropagated()` function will return false.
- * When this occurs, the consumer of this function should retry the `aws` / `gcloud` command.
- *
- * Note that this function requires interception of the subprocess stderr stream.
- * This works because AWS SSM wraps the session in a single-stream pty, so we
- * do not capture stderr emitted from the wrapped shell session.
- */
-const accessPropagationGuard = (
```

Comment:
What's the motivation to remove the sub-functioning for this PR? IMO the code is significantly easier to follow when it's more modular.